dump/plugins/interface.py

```diff
@@ -0,0 +1,273 @@
+        return intf_type
+    
+    def init_intf_appl_info(self):
+        self.add_intf_keys("APPL_DB", "INTF_TABLE")
+    
+    def init_intf_state_info(self):
+        self.add_intf_keys("STATE_DB", "INTERFACE_TABLE")
+    
+    def init_intf_asic_info(self):
+        """
+        Fetch SAI_OBJECT_TYPE_ROUTER_INTERFACE ASIC Object for the corresponding interface
+        To find the relevant ASIC RIF object, this method would need the following:
+        1) INTERFACE - SAI_OBJECT_TYPE_PORT oid
+        2) PORTCHANNEL - SAI_OBJECT_TYPE_LAG oid
+        3) VLAN - SAI_OBJECT_TYPE_VLAN
+        4) SUB_INTERFACE - SAI_OBJECT_TYPE_PORT/SAI_OBJECT_TYPE_LAG & SAI_ROUTER_INTERFACE_ATTR_OUTER_VLAN_ID
+        """
+        rif_obj = RIF.initialize(self)
+        rif_obj.collect()
+        return 
+
+class RIF(object):
+    """
+    Base Class for RIF type
+    """
+    @staticmethod
+    def initialize(intf_obj):
+        if intf_obj.intf_type == "INTERFACE":
+            return PortRIF(intf_obj)
+        elif intf_obj.intf_type == "PORTCHANNEL_INTERFACE":
+            return LagRIF(intf_obj)
+        elif intf_obj.intf_type == "VLAN_INTERFACE":
+            return VlanRIF(intf_obj)
+        elif intf_obj.intf_type == "LOOPBACK_INTERFACE":
+            return LpbRIF(intf_obj)
+        elif intf_obj.intf_type == "VLAN_SUB_INTERFACE":
+            return SubIntfRif(intf_obj)
+        return RIF(intf_obj)
+
+    def __init__(self, intf_obj):
+        self.intf = intf_obj
+    
+    def fetch_rif_keys_using_port_oid(self, port_oid, rfs=["SAI_ROUTER_INTERFACE_ATTR_TYPE"]):
+        if not port_oid:
+            port_oid = "INVALID"
+        req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE", key_pattern="*", field="SAI_ROUTER_INTERFACE_ATTR_PORT_ID",
+              value=port_oid, return_fields=rfs, ns=self.intf.ns)
+        ret = self.intf.match_engine.fetch(req)
+        return req, ret
+
+    def fetch_rif_keys_using_vlan_oid(self, vlan_oid):
+        if not vlan_oid:
+            vlan_oid = "INVALID"
+        req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE", key_pattern="*", field="SAI_ROUTER_INTERFACE_ATTR_VLAN_ID",
+              value=vlan_oid, return_fields=["SAI_ROUTER_INTERFACE_ATTR_TYPE"], ns=self.intf.ns)
+        ret = self.intf.match_engine.fetch(req)
+        return req, ret
+
+    def verify_valid_rif_type(self, ret, exp_rif_type=""):
+        if not ret or not exp_rif_type:
+            return True, ""
+
+        rif_type = ""
+        if not ret["error"] and ret["keys"]:
+            rif_key = ret["keys"][-1]
+            rif_type = ret.get("return_values", {}).get(rif_key, {}).get("SAI_ROUTER_INTERFACE_ATTR_TYPE", "")
+
+        if rif_type == exp_rif_type:
+            return True, rif_type
+        else:
+            return False, rif_type
+
+    def sanity_check_rif_type(self, rif_oid, exp_type, recv_type, str_name):
```

Comment:
Can we combine sanity_check_rif_type and verify_valid_rif_type into a single API?

Reply:
Handled